tests/python_client/testcases/test_index.py

```diff
@@ -1547,6 +1548,261 @@ def test_index_with_string_field_empty(self):
+        collection_w.create_index(default_string_field_name, default_string_index_params, index_name="b")
+        assert collection_w.has_index(index_name="b")[0] == True
+        default_params = {}
+        collection_w.create_index("float", default_params, index_name="c")
+        assert collection_w.has_index(index_name="c")[0] == True
+   
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_drop_diskann_index_with_partition(self):
+        """
+        target: test drop diskann index normal
+        method: 1.create collection and insert data
+                2.create  diskann index and uses collection.drop_index () drop index
+        expected: drop index successfully
+        """
+        c_name = cf.gen_unique_str(prefix)
+        collection_w = self.init_collection_wrap(name=c_name, schema=default_schema)
+        partition_name = cf.gen_unique_str(prefix)
+        partition_w = self.init_partition_wrap(collection_w, partition_name)
+        assert collection_w.has_partition(partition_name)[0]
+        df = cf.gen_default_list_data()
+        ins_res, _ = partition_w.insert(df)
+        collection_w.create_index(default_float_vec_field_name, ct.default_diskann_index)
+        collection_w.load()
+        assert len(collection_w.indexes) == 1
+        collection_w.release()
+        collection_w.drop_index()
+        assert len(collection_w.indexes) == 0
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_create_diskann_index_with_binary(self):
+        """
+        target: test create diskann index with binary
+        method: 1.create collection and insert binary data
+                2.create diskann index 
+        expected: report an error
+        """
+        c_name = cf.gen_unique_str(prefix)
+        collection_w = self.init_collection_wrap(name=c_name, schema=default_binary_schema)
+        df, _ = cf.gen_default_binary_dataframe_data()
+        collection_w.insert(data=df)
+        collection_w.create_index(default_binary_vec_field_name, ct.default_diskann_index, index_name=binary_field_name,
+                                  check_task=CheckTasks.err_res,
+                                  check_items={ct.err_code: 1,
+                                               ct.err_msg: "field data type BinaryVector don't support the index build type DISKANN"})
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_create_diskann_index_multithread(self):
+        """
+        target: test create index interface with multiprocess
+        method: create collection and add entities in it, create diskann index
+        expected: return search success
+        """
+        c_name = cf.gen_unique_str(prefix)
+        collection_w = self.init_collection_wrap(name=c_name)
+        data = cf.gen_default_list_data(default_nb)
+        collection_w.insert(data=data)
+
+        def build(collection_w):
+
+            collection_w.create_index(ct.default_float_vec_field_name, default_index_params)
```

Comment:
Here default_index_params is not diskann index?

Reply:
Formatting has been changed